src/core/qgsabstractdatabaseproviderconnection.h

```diff
@@ -0,0 +1,406 @@
+     *
+     * In case the table is a vector spatial table and it has multiple
+     * geometry columns, separate entries for each geometry column must
+     * be created.
+     *
+     * In case the table is a vector spatial table and the geometry column
+     * can contain multiple geometry types and/or SRIDs, a clone of the property
+     * for the individual geometry type/SRID can be retrieved with at(i)
+     */
+    struct TableProperty
+    {
+
+#ifdef SIP_RUN
+        SIP_PYOBJECT __repr__();
+        % MethodCode
+        QString str = QStringLiteral( "<QgsAbstractDatabaseProviderConnection.TableProperty: '%1'>" ).arg( sipCpp->tableName() );
+        sipRes = PyUnicode_FromString( str.toUtf8().constData() );
+        % End
+#endif
+
+      public:
+
+        /**
+         * Returns the table name
+         * \see defaultName()
+         */
+        QString tableName() const;
+
+        /**
+         * Sets the table name to \a name
+         * \see defaultName()
+         */
+        void setTableName( const QString &name );
+
+        /**
+         * Appends the geometry column \a type with the given \a srid to the geometry column types list
+         */
+        void addGeometryType( const QgsWkbTypes::Type &type, const int srid );
+
+        /**
+         * Returns the list of geometry column types and SRIDs.
+         * The method returns a list of tuples (QgsWkbTypes::Type, SRID)
+         */
+        QList<QPair<int, int>> geometryTypes() const;
+
+        /**
+         * Returns the default name for the table entry
+         *
+         * It is usually the table name but in case there are multiple geometry
+         * columns, the geometry column name is appendend to the table name.
+         * \see geometryColumnCount()
+         */
+        QString  defaultName() const;
+
+        /**
+         * Returns the table property corresponding to the geometry type a
+         * the given indext \a index
+         */
+        TableProperty at( int index ) const;
+
+        /**
+         * Returns the schema or an empty string for backends that do not support a schema
+         */
+        QString schema() const;
+
+        /**
+         * Sets the \a schema
+         */
+        void setSchema( const QString &schema );
+
+        /**
+         * Returns the geometry column name
+         */
+        QString geometryColumn() const;
+
+        /**
+         * Sets the geometry column name to \a geometryColumn
+         */
+        void setGeometryColumn( const QString &geometryColumn );
+
+        /**
+         * Returns the list of primary key column names
+         */
+        QStringList pkColumns() const;
+
+        /**
+         * Sets the primary key column names to \a pkColumns
+         */
+        void setPkColumns( const QStringList &pkColumns );
+
+        /**
+         * Returns the list of SRIDs supported by the geometry column
+         */
+        QList<int> srids() const;
```

Comment:
again, replace with a list of auth:id codes (or QgsCoordinateReferenceSystem objects)

Reply:
As mentioned above, I don't think I can do it now unless I wanted to change the provider code too.